src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultActionModelBuilder.cs

```diff
@@ -266,6 +270,22 @@ private bool IsIDisposableMethod(MethodInfo methodInfo, TypeInfo typeInfo)
             AddRange(actionModel.ActionConstraints, attributes.OfType<IActionConstraintMetadata>());
             AddRange(actionModel.Filters, attributes.OfType<IFilter>());
 
+            var enableCors = attributes.OfType<IEnableCorsMetadata>().SingleOrDefault();
+            if (enableCors != null)
+            {
+                var corsPolicy = _corsOptions.GetPolicy(enableCors.PolicyName);
```

Comment:
Should the policy be retrieved from an implementation of ICorsPolicyProvider instead (in this case attribute-based policy provider which looks at corsoptions)...a user could replace this policy provider if they want to...

example: custom policy provider which could be attribute-based but builds the policy from information in database...